dataflow/src/org/checkerframework/dataflow/cfg/node/LambdaResultExpressionNode.java

```diff
@@ -0,0 +1,81 @@
+package org.checkerframework.dataflow.cfg.node;
+
+import com.sun.source.tree.ExpressionTree;
+import java.util.Collection;
+import java.util.Collections;
+import javax.lang.model.util.Types;
+import org.checkerframework.dataflow.util.HashCodeUtils;
+import org.checkerframework.javacutil.InternalUtils;
+
+/*>>>
+import org.checkerframework.checker.nullness.qual.Nullable;
+*/
+
+/**
+ * A node for the single expression body of a single expression lambda
+ *
+ * @author Lazaro Clapp
+ */
+public class LambdaResultExpressionNode extends Node {
+
+    protected ExpressionTree tree;
+    protected /*@Nullable*/ Node result;
+
+    public LambdaResultExpressionNode(
+            ExpressionTree t, /*@Nullable*/ Node result, Types types /*,
+            LambdaExpressionTree lambda,
+            Symbol.MethodSymbol methodSymbol*/) {
+        super(InternalUtils.typeOf(t));
+        this.result = result;
+        tree = t;
+        //result.setAssignmentContext(new AssignmentContext.LambdaReturnContext(methodSymbol));
+    }
+
+    public Node getResult() {
+        return result;
+    }
+
+    @Override
+    public ExpressionTree getTree() {
+        return tree;
+    }
+
+    @Override
+    public <R, P> R accept(NodeVisitor<R, P> visitor, P p) {
+        return visitor.visitLambdaResultExpression(this, p);
+    }
+
+    @Override
+    public String toString() {
+        if (result != null) {
+            return "-> " + result;
+        }
+        return "-> ()";
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null || !(obj instanceof LambdaResultExpressionNode)) {
+            return false;
+        }
+        LambdaResultExpressionNode other = (LambdaResultExpressionNode) obj;
+        if ((result == null) != (other.result == null)) {
+            return false;
+        }
```

Comment:
Can you add a comment why the `tree` doesn't need to be compared?

Reply:
You know what, let me add a comparison for `tree` too. It shouldn't matter if we are constructing `LambdaResultExpressionNode`'s correctly always (`result` is supposed to be `scan([path_to_tree], null);`, but we are not enforcing this within the class, so lets both check here and use it to compute the hash)

Reply:
Although, actually, `ReturnNode`, which is what I am basing this class on, doesn't use `tree` as part of its comparison... mmm...